### PR TITLE
chore: allure integration and migration utility for component tests

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?
Currently the component tests are not very easy to navigate and well documented. This PR aims to bridge tho following issues

Incorrect or no usage of type systems in the test

A migration utility has been introduced to migrate tests which to correct types

Upon migration the user is expected to fill the details of the test cases like descriptions and the scenario details

Visualisation problem:  Right now we have visualisation tools for coverage but not to visualise and understand test cases this results in difficulties in navigating through tests cases during issues or CFD

Integration of [allure](https://allurereport.org/) test reporter for solving test visualisation and navigation

## What is the related Linear task?

Resolves INT-2941

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by utsabc** [comment]
Callout: Since `method` is added to test types hence needed to manually add to all test cases where it was missing

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3949/test-report.html

File: test/scripts/migrateTest.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+    // Write the formatted content
+    fs.writeFileSync(filePath, formattedContent);
+  } catch (error) {
+    console.error(`Error writing to file ${filePath}:`, error);
+    throw error;
+  }
+}
+
+async function enhwriteTestFile(
```
Comment (koladilip) [inline test/scripts/migrateTest.ts:78]: rename for better clarity

**Review by koladilip — DISMISSED** [review]

File: test/scripts/migrateTest.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    return formattedContent;
+  } catch (error) {
+    console.error('Error formatting file with prettier:', error);
+    // Return original content if formatting fails
+    return content;
+  }
+}
+
+async function writeTestFile(filePath: string, data: any): Promise<void> {
```
Comment (krishna2020) [inline test/scripts/migrateTest.ts:57]: Is this function being used ?

Comment (utsabc) [reply test/scripts/migrateTest.ts]: removed

File: test/scripts/migrateTest.ts (hunk 1 above)
Comment (utsabc) [reply test/scripts/migrateTest.ts]: fixed

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**